### DISCUSSION TITLE
Difference between the times

### DISCUSSION THREAD

**Discussion by BolunDai0216** [Simple-Robotics/proxsuite#223] [General]
Hi,

When checking the solver results I get 3 different time measurements: `run_time`, `setup_time`, and `solve_time`. Is it possible to help me understand what they are referring to?

**Comment by jcarpent** [comment]
@Bambade Could you provide a bit of guidance here?

**Comment by Bambade** [comment]
Hi @BolunDai0216,

Thank you for your question ! Please find below a description of what they refer to:
- **setup_time**: measures the time required for initializing the model (e.g., for allocating memory),
- **solve_time**: measures everything else (i.e., solving the QP),
- **run_time** = setup_time + solve_time.

If need be you can find other information in [our documentation](https://simple-robotics.github.io/proxsuite/md_doc_2_PROXQP_API_2_ProxQP_api.html#OverviewResults). 

Feel free to ask if you have other questions
